operator/scheduler/client.go

```diff
@@ -66,30 +67,68 @@ func getSchedulerHost(namespace string) string {
 	return fmt.Sprintf("seldon-scheduler.%s", namespace)
 }
 
+// startEventHanders starts the grpc stream connections to the scheduler for the different resources we care about
+// we also add a retry mechanism to reconnect if the connection is lost, this can happen if the scheduler is restarted
+// or if the network connection is lost. We use an exponential backoff to retry the connection.
+// note that when the scheduler is completely dead we will be not be able to reconnect and these go routines will retry forever
+// TODO: add a max retry count and report back to the caller.
 func (s *SchedulerClient) startEventHanders(namespace string, conn *grpc.ClientConn) {
+	retryFn := func(fn func(context context.Context, conn *grpc.ClientConn) error, context context.Context, conn *grpc.ClientConn) error {
+		logFailure := func(err error, delay time.Duration) {
+			s.logger.Error(err, "Scheduler not ready")
```

Comment:
One of the issues here is that we'll probably be seeing each error 4 times when the scheduler is down (once from each of the coroutines below), but I'm not sure if we can sort that out easily.